src/MentionsInput.js

```diff
@@ -241,30 +234,31 @@ class MentionsInput extends React.Component {
     }
   }
 
+  setSuggestionsElement = (el) => {
+    this.suggestionsElement = el
+  }
+
   renderSuggestionsOverlay = () => {
     if (!isNumber(this.state.selectionStart)) {
       // do not show suggestions when the input does not have the focus
       return null
     }
 
+    const { position, left, top } = this.state.suggestionsPosition
+
     const suggestionsNode = (
       <SuggestionsOverlay
         style={this.props.style('suggestions')}
-        position={this.state.suggestionsPosition}
+        position={position}
+        left={left}
+        top={top}
         focusIndex={this.state.focusIndex}
         scrollFocusedIntoView={this.state.scrollFocusedIntoView}
-        ref={el => {
-          this.suggestionsRef = el
-        }}
+        containerRef={this.setSuggestionsElement}
         suggestions={this.state.suggestions}
         onSelect={this.addMention}
         onMouseDown={this.handleSuggestionsMouseDown}
-        onMouseEnter={focusIndex =>
-          this.setState({
-            focusIndex,
-            scrollFocusedIntoView: false,
-          })
-        }
+        onMouseEnter={this.handleSuggestionsMouseEnter}
```

Comment:
here we're potentially avoiding some unnecessary re-renders (similar optimizations have been applied in other places)